syntax = "proto3";

package videoUpload;


// The message representing a chunk of video data (binary content).
message VideoChunk {
  bytes data = 1; // A chunk of the video file data (a portion of the video)
  int64 chunk_index = 2; // The index of this chunk (helps with reordering)
  bool is_last_chunk = 3; // Indicates if this is the last chunk of the video
  string chunk_type = 4; // for streaming to video player
}

message VideoRequest {
  string video_id = 1;
}

message Status{
  string status = 1;          // Status of the upload (e.g., "success", "failure")
  string message = 2;         // Additional message (e.g., error message if any)
  int64 progress = 3;
}
// Response for the video upload.
message UploadVideoResponse {
  Status status = 1;
}

// Video upload service definition with streaming.
service VideoUploadService {
  // RPC method for streaming video data and uploading it.
  rpc UploadVideo (stream VideoChunk) returns (stream UploadVideoResponse);
  rpc StreamVideo (VideoRequest) returns (stream VideoChunk);

}